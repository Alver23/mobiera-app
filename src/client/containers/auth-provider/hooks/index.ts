// Dependencies
import React from 'react';

// Commons
import {
  clearAuthData,
  getAuthData,
} from '@mobiera/containers/login-form/commons';

// Interfaces
import { IUser } from '@mobiera/services/authentication/interfaces';
import IMutableRefObject from '@mobiera/interfaces/mutable-ref';
import IAuthContext from '../context/interfaces';

// Contexts
import AuthContext from '../context';

const useAuth = (): [IAuthContext, () => void, (values: any) => void] => {
  const initialState: IAuthContext = React.useMemo(
    () => ({
      authenticated: false,
      initialize: false,
    }),
    []
  );

  const [data, setData] = React.useState<IAuthContext>(initialState);
  const isMounted: IMutableRefObject<boolean> = React.useRef(true);

  React.useEffect(() => {
    return (): void => {
      isMounted.current = false;
    };
  }, []);

  const logout = React.useCallback((): void => {
    setData({ ...initialState, initialize: true });
    clearAuthData().then();
  }, [initialState]);

  const updateData = React.useCallback((newValues): void => {
    setData((prevState) => ({
      ...prevState,
      ...newValues,
    }));
  }, []);

  React.useEffect((): void => {
    const setAuthData = (user: IUser) => {
      if (isMounted.current) {
        setData({
          user,
          initialize: true,
          authenticated: !!user,
        });
      }
    };

    const setError = () => {
      if (isMounted.current) {
        setData(initialState);
      }
    };

    getAuthData().then(setAuthData).catch(setError);
  }, [initialState]);

  return [data, logout, updateData];
};

export const useAuthSession = (): IAuthContext => {
  return React.useContext<IAuthContext>(AuthContext);
};

export default useAuth;
