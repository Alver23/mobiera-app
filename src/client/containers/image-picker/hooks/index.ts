// Dependencies
import React from 'react';

const useImagePicker = (): [any, () => void] => {
  const actionSheetRef: any = React.useRef();
  const openImagePicker = React.useCallback(() => {
    actionSheetRef.current.show();
  }, []);

  return [actionSheetRef, openImagePicker];
};

export default useImagePicker;
