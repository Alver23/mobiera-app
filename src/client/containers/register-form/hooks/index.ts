// Dependencies
import React from 'react';

// Hooks
import useShowAppLoader from '@mobiera/hooks/app-loader';

// Services
import UserServices from '@mobiera/services/user';

// Interfaces
import { IFormData } from '@mobiera/containers/user-form/interfaces';
import { IFormData as IPayload } from '../interfaces';

const useSignUp = (): [string | undefined, (payload: IFormData) => void] => {
  const [showAppLoader] = useShowAppLoader();
  const [message, setMessage] = React.useState<string>();

  const onSubmit = React.useCallback(
    (payload: IFormData) => {
      const onSuccess = () => {
        setMessage('The user was created successfully');
        showAppLoader(false);
      };

      const onError = () => {
        setMessage('An error has occurred, please try again later');
        showAppLoader(false);
      };

      showAppLoader(true);
      UserServices.save(payload as IPayload)
        .then(onSuccess)
        .catch(onError);
    },
    [showAppLoader]
  );

  return [message, onSubmit];
};

export default useSignUp;
