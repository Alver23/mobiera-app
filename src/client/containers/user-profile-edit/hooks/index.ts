// Dependencies
import React from 'react';

// Hooks
import useShowAppLoader from '@mobiera/hooks/app-loader';

// Services
import UserServices from '@mobiera/services/user';

// Interfaces
import { IFormData } from '@mobiera/containers/user-form/interfaces';
import { IUserUpdateRequest } from '@mobiera/services/user/interfaces';

const useUserUpdate = (
  id: number,
  callback: () => void,
  updateData: (values: any) => void
): [string | undefined, (payload: IFormData) => void] => {
  const [showAppLoader] = useShowAppLoader();
  const [message, setMessage] = React.useState<string>();

  const onSubmit = React.useCallback(
    (payload: IFormData) => {
      const onSuccess = () => {
        setMessage('The user was updated successfully');
        showAppLoader(false);
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { password, ...userPayload } = payload;
        updateData({ user: userPayload });
        callback();
      };

      const onError = () => {
        setMessage('An error has occurred, please try again later');
        showAppLoader(false);
      };

      showAppLoader(true);
      UserServices.update(id, payload as IUserUpdateRequest)
        .then(onSuccess)
        .catch(onError);
    },
    [id, showAppLoader, callback, updateData]
  );

  return [message, onSubmit];
};

export default useUserUpdate;
