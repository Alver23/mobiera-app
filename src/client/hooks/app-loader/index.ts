// Dependencies
import { useCallback } from 'react';
import { useDispatch } from 'react-redux';

// Redux Actions
import appShowLoaderAction from '@mobiera/store/actions/globals';

const useShowAppLoader = (): [(show: boolean) => void] => {
  const dispatch = useDispatch();
  const callback = useCallback(
    (show: boolean) => {
      return dispatch(appShowLoaderAction(show));
    },
    [dispatch]
  );

  return [callback];
};

export default useShowAppLoader;
