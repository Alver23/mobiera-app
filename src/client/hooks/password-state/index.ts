// Dependencies
import React from 'react';

const usePasswordState = (
  initialState = true
): [string, boolean, () => void] => {
  const [showPassword, setShowPassword] = React.useState<boolean>(initialState);
  const iconsMapper: any = {
    true: 'eye',
    false: 'eye-slash',
  };
  const icon = iconsMapper[`${showPassword}`];

  const onChangeSetPassword = React.useCallback(() => {
    setShowPassword(!showPassword);
  }, [showPassword]);

  return [icon, showPassword, onChangeSetPassword];
};

export default usePasswordState;
