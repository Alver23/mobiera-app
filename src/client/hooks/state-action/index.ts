// Dependencies
import React from 'react';

const useStateAction = (
  initialState?: boolean
): [boolean, () => void, () => void] => {
  const [state, setState] = React.useState(!!initialState);
  const truthyCallback = React.useCallback((): void => setState(true), []);
  const falsyCallback = React.useCallback((): void => setState(false), []);
  return [state, truthyCallback, falsyCallback];
};

export default useStateAction;
