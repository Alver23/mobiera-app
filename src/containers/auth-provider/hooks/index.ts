// Dependencies
import React from 'react';

// Interfaces
import IMutableRefObject from '@mobiera/interfaces/mutable-ref';
import IAuthContext from '@mobiera/containers/auth-provider/context/interfaces';

const useAuth = (): [IAuthContext, () => void] => {
  const initialState: IAuthContext = React.useMemo(
    () => ({
      authenticated: false,
      initialize: false,
    }),
    []
  );

  const [data, setData] = React.useState<IAuthContext>(initialState);
  const isMounted: IMutableRefObject<boolean> = React.useRef(true);

  React.useEffect(() => {
    return (): void => {
      isMounted.current = false;
    };
  }, []);

  const logout: () => void = React.useCallback((): void => {
    setData({ ...initialState, initialize: true });
  }, [initialState]);

  React.useEffect((): void => {
    setData((prevstate) => ({
      ...prevstate,
      initialize: true,
    }));
  }, []);

  return [data, logout];
};

export default useAuth;
