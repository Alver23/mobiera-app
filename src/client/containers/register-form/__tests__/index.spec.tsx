// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

// Under test file
import RegisterForm from '../index';

// Hooks
import * as signUpHooks from '../hooks';

describe('<RegisterForm />', () => {
  let component: ShallowWrapper;
  const signUpMock: [string, jest.Mock] = ['message', jest.fn()];

  beforeEach(() => {
    jest.spyOn(signUpHooks, 'default').mockReturnValue(signUpMock);
    component = shallow(<RegisterForm />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });
});
