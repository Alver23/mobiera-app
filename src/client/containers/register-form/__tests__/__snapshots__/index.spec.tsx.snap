// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterForm /> should save a snapshot of the component 1`] = `
<View
  style={
    Object {
      "marginVertical": 8,
    }
  }
>
  <UserForm
    formSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_blacklistError": undefined,
        "_excludedEdges": Array [],
        "_nodes": Array [
          "password",
          "email",
          "name",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_whitelistError": undefined,
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "email": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "email": false,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "name": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "password": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "matches": false,
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
    onSubmit={[MockFunction]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 8,
        }
      }
    >
      <Themed.Text
        style={
          Object {
            "color": "#ff190c",
          }
        }
      >
        message
      </Themed.Text>
    </View>
  </UserForm>
</View>
`;
